tools/compute_accuracy.py

```diff
@@ -0,0 +1,238 @@
+        if oversample:
+            predictions = predictions.reshape((len(predictions) / 10, 10, -1))
+            predictions = predictions.mean(1)
+
+        return predictions
+
+def compute_accuracy(snapshot, deploy_file, labels_file, mean_file, img_set, width, height, resize_mode, grayscale=False, oversample=False):
+    """
+    Evaluate a Net on a set of images, and dump the result in two
+    pickle files.
+    Returns True on sucess
+
+    Arguments:
+    snapshot -- a Caffe trained model
+    deploy_file -- the corresponding deploy file
+    labels_file -- the file containing the dataset labels
+    mean_file -- the dataset mean file
+    img_set -- the file containing a list of images
+    resize_mode -- the mode used to resize images
+
+    Keyword arguments:
+    oversampling -- boolean, True if oversampling should be enabled
+    """
+
+    img_matrix = pd.read_csv(img_set, engine='python', header=None,sep=r" (?!(.*) )")
+
+    mean_blob = caffe.proto.caffe_pb2.BlobProto()
+    mean_data = open(mean_file, 'rb').read()
+    mean_blob.ParseFromString(mean_data)
+    mean = np.array(caffe.io.blobproto_to_array(mean_blob))[0].mean(1).mean(1)
+
+    # Loading the classifier
+    caffe.set_mode_gpu()
+    if grayscale:
+        net = Classifier(deploy_file, snapshot,
+                       mean=mean,
+                       raw_scale=255)
+    else:
+        net = Classifier(deploy_file, snapshot,
+                       mean=mean,
+                       channel_swap=(2,1,0),
+                       raw_scale=255)
+
+    size = len(img_matrix)
+    num_classes = np.array(img_matrix[2])[-1]+1
+    prediction = np.zeros(size, dtype=int)
+    labels = np.zeros(size, dtype=int)
+    probas = np.zeros([size, num_classes], dtype=float)
+ 
+    for i in range(0, size): 
+        cur_image = img_matrix[0][i]
+        if grayscale:
+            input_image = utils.image.load_image(cur_image)            
+            input_image = utils.image.resize_image(input_image, width, height, resize_mode=resize_mode, channels=1)
+            input_image = input_image.astype(np.uint8)
+            input_image = input_image.reshape(input_image.shape[0], input_image.shape[1], 1)
+        else:
+            input_image = utils.image.load_image(cur_image)
+            input_image = utils.image.resize_image(input_image, width, height, resize_mode=resize_mode)
+            input_image = skimage.img_as_float(input_image).astype(np.float32)
+
+        labels[i] = img_matrix[2][i]
+        probas[i] = net.predict([input_image], oversample=False)
+        prediction[i] = probas[i].argmax()
+        if size > 500 and i % (size/500) == 0:
+            logger.debug("Progress: %0.2f" % (i/float(size)))
+
+    snapshot_file, snapshot_extension = os.path.splitext(snapshot)
+
+    # Dumping the result
+    joblib.dump(probas, snapshot_file + "-accuracy-proba.pkl")
+    joblib.dump(labels, snapshot_file + "-accuracy-labels.pkl")
```

Comment:
Can we use [Pickle](https://docs.python.org/2/library/pickle.html) instead of joblib to avoid adding a new dependency? See usage [here](https://github.com/NVIDIA/DIGITS/blob/30fde0264f2b5dd84d3844c3cc69c92044d968a7/digits/job.py#L177-L178) and [here](https://github.com/NVIDIA/DIGITS/blob/30fde0264f2b5dd84d3844c3cc69c92044d968a7/digits/job.py#L33-L34).